In [2]:
import numpy as np
import astropy.units as u
import astropy.constants as con
import matplotlib.pyplot as plt
%matplotlib inline

The Herschel GBS group seems to follow the route of [Könyves et al. (2010)](http://adsabs.harvard.edu/abs/2010A%26A...518L.106K), which in turn cites [Beckwith et al. (1990)](http://adsabs.harvard.edu/abs/1990AJ.....99..924B) for the opacity law: $\kappa_{\nu} = 0.1 \mathrm{cm^2~g^{-1}} (\nu/\mathrm{1~THz})^\beta$ and [Hildebrand (1983)](http://adsabs.harvard.edu/abs/1983QJRAS..24..267H) for $\beta=2$.  

Radiative transfer gives us that $I_{\nu}(\nu) = B_{\nu}(\nu,T) [1-\exp(-\tau_\nu)] = B_{\nu}(\nu,T) [1-\exp(-\kappa_\nu\Sigma)]$. Solving for $\Sigma$ gives

$$ \Sigma_{\mathrm{dust}} = \frac{I_\nu(\nu)}{B_\nu(\nu,T)\kappa_\nu}$$
and $\Sigma_{\mathrm{gas}} = \Gamma \Sigma_{\mathrm{dust}}$ where $\Gamma = 100$ is the canonical (i.e., I have to find a reference) gas-to-dust ratio.  Note that the $\kappa_\nu$ quoted above already includes the $\Gamma$ factor, which should probably be stated in the Köynves article somewhere...

In [3]:
def bnu(nu,T):
    bnu = 2*con.h*nu**3/con.c**2*(1/(np.exp(con.h*nu/(con.k_B*T))-1))/u.sr
    return bnu.to(u.MJy/u.sr)

In [4]:
lam = np.array([250,350,500])*u.um
nu = (con.c/lam).to(u.GHz)
beta = 2.0
kappa = 0.1*(nu/u.THz)**beta * u.cm**2/u.g
kappa.to(u.cm**2/u.g)

<Quantity [ 0.14380083, 0.07336777, 0.03595021] cm2 / g>

In [5]:
((bnu(nu,15*u.K)*kappa.to(u.pc**2/u.Msun)))**(-1)

<Quantity [ 0.59402307, 1.02060932, 2.43394581] solMass sr / (MJy pc2)>

Thus, at 350 $\mu$m, assuming $T_{\mathrm{dust}}=15$ K the conversion between intensity and mass is 

$$\Sigma_{\mathrm{gas}} = 1.02~M_{\odot}~\mathrm{pc}^{-2} \left[\frac{I_{\nu}(\nu)}{\mathrm{MJy~sr}^{-1}}\right]$$

In [6]:
Tgas = 10*u.K
mu = 2.38
cs = ((con.k_B*Tgas/(mu*con.m_p))**0.5).to(u.m/u.s)
lam = (2*cs**2/con.G).to(u.Msun/u.pc)

In [7]:
lam

<Quantity 16.12340075655668 solMass / pc>

Chen and Ostriker give the critical mass for a cylinder as 
$$ 14 \frac{c_s}{\sqrt{G^3} B_{\mathrm{ps}}}$$

In [8]:
Bps = 3e-5*(u.g**0.5/u.s/u.cm**0.5)
Mcrit = 14*cs**4/(con.G**1.5*Bps)
W=0.1*u.pc

In [9]:
Mcrit.to(u.Msun)

<Quantity 1.6368297480819811 solMass>

In [10]:
LamMag = Bps*W/(2*np.pi*con.G**0.5)

In [11]:
LamMag.to(u.Msun/u.pc)

<Quantity 8.847039455537114 solMass / pc>